dandiapi/api/views/asset.py

```diff
@@ -511,14 +511,18 @@ def destroy(self, request, versions__dandiset__pk, versions__version, **kwargs):
 
         return Response(None, status=status.HTTP_204_NO_CONTENT)
 
-    @swagger_auto_schema(query_serializer=AssetListSerializer)
+    @swagger_auto_schema(query_serializer=AssetListSerializer, responses={200: AssetSerializer()})
     def list(self, request, *args, **kwargs):
         serializer = AssetListSerializer(data=request.query_params)
         serializer.is_valid(raise_exception=True)
 
-        queryset = self.filter_queryset(self.get_queryset())
-        glob_pattern: str | None = serializer.validated_data.get('glob')
+        # Don't include metadata field if not asked for
+        queryset: QuerySet[Asset] = self.filter_queryset(self.get_queryset().select_related('blob'))
```

Comment:
just to make sure -- that `.select_related('blob')` wouldn't just select assets with blobs thus leaving behind zarr assets, would it?  not clear to me why that change is done for this PR

and shouldn't comment above that line be moved below this line?

Reply:
`select_related('blob')` joins the `blob` foreign key into the single query, so that we don't need to fetch `blob` from the database individually for every asset when it's serialized into the response (causing n queries where n is the number of assets). It's a performance optimization.

Reply:
Is that change necessary to the purpose of the PR (adding metadata) or just a general optimization?

Reply:
Just a general optimization, the codebase is littered with unoptimized endpoints.

Reply:
then ideally should have been done at least in a separate commit with an explanation alike given [above](https://github.com/dandi/dandi-archive/pull/1216#discussion_r930467945) for others to be able to understand why it is done in a commit message.
Otherwise poor souls like yours truly would spend numerous hours while trying to understand why it was done in the context of the "add metadata subrecord" -- e.g. "was blob somehow needed for metadata?".

tip of trade: I use emacs only for magit mode which allows for convenient staging of hunts and rewrite of commits. Otherwise, could be done "manually" via `git add -i` and choosing one hunk at a time.

re "litter" -- filed a dedicated https://github.com/dandi/dandi-archive/issues/1221 . Let's try to keep PRs and commits atomic regarding their purpose.

PS with this I consider this conversation indeed resolved.